source/code/plugins/in_npmd_server.rb

```diff
@@ -125,6 +127,8 @@ def start
             @npmdIntendedStop = false
             @stderrFileNameHash = Hash.new
             @stop_sync = Mutex.new
+            @networkAgentState = nil
+            @netWorkAgentSync = Mutex.new
```

Comment:
Might not need this as per code here. The npmd_reader is single thread of execution and I don't see state being altered anywhere else.